piet-direct2d/src/dwrite.rs

```diff
@@ -133,42 +151,114 @@ impl DwriteFactory {
     }
 }
 
-impl TextFormatBuilder {
-    pub fn new(factory: DwriteFactory) -> TextFormatBuilder {
-        TextFormatBuilder {
-            factory,
-            size: None,
-            family: None,
-        }
-    }
+impl FontCollection {
+    pub(crate) fn font_family(&self, name: &str) -> Option<FamilyName> {
+        let wname = name.to_wide_null();
+        let mut idx = u32::max_value();
+        let mut exists = 0_i32;
+
+        let family = unsafe {
+            let hr = self.0.FindFamilyName(wname.as_ptr(), &mut idx, &mut exists);
+            if SUCCEEDED(hr) && exists != 0 {
+                let mut family = null_mut();
+                let hr = self.0.GetFontFamily(idx, &mut family);
+                wrap(hr, family, FontFamily).ok()
+            } else {
+                eprintln!(
+                    "failed to find family name {}: err {} not_found: {}",
+                    name, hr, !exists
+                );
+                None
+            }
+        }?;
 
-    pub fn size(mut self, size: f32) -> TextFormatBuilder {
-        self.size = Some(size);
-        self
+        family.family_name().ok()
     }
+}
+
+impl FontFamily {
+    // this monster is taken right out of the docs :/
+    /// Returns the localized name of this family.
```

Comment:
Getting localized names feels really wrong to me, though I'm willing to accept that it is the best of not-so-good options for identifying a font. I'm curious whether there is other precedent for this (browsers, font-kit).

I did look at the documentation for `GetFamilyNames` and accept that this is a good implementation :)

Reply:
I've debated taking this out, and instead just using whatever name we've been given if it returns a family; we can trust that name will return a family in the future, and there's no advantage to storing the canonical name (that i know of) and getting it requires us to jump through all of these hoops. Ultimately I decided to just leave this here since it's "more correct" and it already works, but I could go either way.